src/shogun2-core/shogun2-service/src/main/java/de/terrestris/shogun2/service/GeoServerInterceptorService.java

```diff
@@ -0,0 +1,611 @@
+package de.terrestris.shogun2.service;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Properties;
+import java.util.TreeMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpException;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.entity.ContentType;
+import org.apache.http.message.BasicNameValuePair;
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.http.HttpHeaders;
+import org.springframework.stereotype.Service;
+import org.w3c.dom.Document;
+
+import de.terrestris.shogun2.model.interceptor.InterceptorRule;
+import de.terrestris.shogun2.util.http.HttpUtil;
+import de.terrestris.shogun2.util.interceptor.InterceptorException;
+import de.terrestris.shogun2.util.interceptor.MutableHttpServletRequest;
+import de.terrestris.shogun2.util.interceptor.OgcMessage;
+import de.terrestris.shogun2.util.interceptor.OgcMessageDistributor;
+import de.terrestris.shogun2.util.interceptor.OgcNaming;
+import de.terrestris.shogun2.util.interceptor.OgcXmlUtil;
+import de.terrestris.shogun2.util.model.Response;
+
+/**
+ *
+ * @author Daniel Koch
+ * @author Kai Volland
+ * @author terrestris GmbH & Co. KG
+ *
+ */
+@Service
+public class GeoServerInterceptorService {
+
+	/**
+	 * The Logger.
+	 */
+	private static final Logger LOG = Logger.getLogger(
+			GeoServerInterceptorService.class);
+
+	/**
+	 * The autowired properties file containing the (application driven)
+	 * GeoServer namespace - GeoServer BaseURI mapping, e.g.:
+	 *
+	 *   topp=http://localhost:8080/geoserver/topp/ows
+	 *
+	 */
+	private Properties geoServerNameSpaces;
+
+	/**
+	 * An array of whitelisted Headers to forward within the Interceptor.
+	 */
+	private static final String[] FORWARD_HEADER_KEYS = new String[] {
+		"Content-Type",
+		"Content-Disposition",
+		"geowebcache-cache-result",
+		"geowebcache-crs",
+		"geowebcache-gridset",
+		"geowebcache-tile-bounds",
+		"geowebcache-tile-index"
+	};
```

Comment:
I would prefer to not make any assumptions here and forward any headers which don't cause problems. I know this will be done by a dedicated proxy class in the future.